src/3d/materials/qgsgoochmaterialsettings.cpp

```diff
@@ -113,3 +127,143 @@ Qt3DRender::QMaterial *QgsGoochMaterialSettings::toMaterial( QgsMaterialSettings
 void QgsGoochMaterialSettings::addParametersToEffect( Qt3DRender::QEffect * ) const
 {
 }
+
+QByteArray QgsGoochMaterialSettings::dataDefinedVertexColorsAsByte( const QgsExpressionContext &expressionContext ) const
+{
+
+  QColor diffuse = dataDefinedProperties().valueAsColor( Diffuse, expressionContext, mDiffuse );
+  QColor warm = dataDefinedProperties().valueAsColor( Warm, expressionContext, mWarm );
+  QColor cool = dataDefinedProperties().valueAsColor( Cool, expressionContext, mCool );
+  QColor specular = dataDefinedProperties().valueAsColor( Specular, expressionContext, mSpecular );
+
+
+  QByteArray array;
+  array.resize( sizeof( float ) * 12 );
```

Comment:
do we need to use 4-byte float for each color component? why not just 1-byte `unsigned char` for each color component? The colors will be translated to 1-byte values in the final R,G,B color anyway...